toolkit/tools/srpmpacker/srpmpacker.go

```diff
@@ -881,7 +885,12 @@ func hydrateFiles(fileTypeToHydrate fileType, specFile, workingDir string, srcCo
 	}
 
 	if hydrateRemotely && srcConfig.sourceURL != "" {
-		hydrateFromRemoteSource(fileHydrationState, newSourceDir, srcConfig, skipSignatureHandling, currentSignatures, cancel, netOpsSemaphore)
+		err = hydrateFromRemoteSource(fileHydrationState, newSourceDir, srcConfig, skipSignatureHandling, currentSignatures, cancel, netOpsSemaphore)
+		// Most errors are non-fatal and are handled inside the function (encoded inside fileHydrationState), cancellation
+		// however will return and error that should cause an early exit.
```

Comment:
I'm not sure we want such comment outside the function itself. They tend to get stale easily after a function update, since they are hard to keep track of. I'd say we might need it - the main point is we return fast in case of an error without diving into what caused it.